cobbler/cobbler_collections/collection.py

```diff
@@ -256,33 +256,38 @@ def rename(
         # Save the old name
         oldname = ref.name
         # Reserve the new name
-        self.listing[newname] = None
+        self.listing[newname.lower()] = None
```

Comment:
I have nothing against that change but then we should document that only lowercase names are supported. As the name is currently the primary key this could lead to ambiguous search results and duplicate dictionary keys afaik.

Reply:
In fact, in the main branch, collection keys are always converted to lowercase when a new collection element is added:
https://github.com/cobbler/cobbler/blob/3da0a3236fc058f66e413fc595f75478bd91485e/cobbler/cobbler_collections/collection.py#L424-L425
Therefore, it is implicitly assumed that cobbler does not distinguish the case of names. This is taken into account everywhere in the code except for `rename`:
https://github.com/cobbler/cobbler/blob/3da0a3236fc058f66e413fc595f75478bd91485e/cobbler/cobbler_collections/collection.py#L95-L102
https://github.com/cobbler/cobbler/blob/3da0a3236fc058f66e413fc595f75478bd91485e/cobbler/cobbler_collections/collection.py#L131-L136
It seems to me that in `rename` simply forgot to do this, and only in this operation an error occurs if the collection name contains uppercase characters:
```
File "/usr/lib/python3.11/site-packages/cobbler/remote.py", line 2151, in xapi_object_edit
self.rename_item(object_type, handle, attributes["newname"], token)
File "/usr/lib/python3.11/site-packages/cobbler/remote.py", line 1657, in rename_item
self.api.rename_item(what, obj, newname)
File "/usr/lib/python3.11/site-packages/cobbler/api.py", line 843, in rename_item
self.get_items(what).rename(ref, newname)
File "/usr/lib/python3.11/site-packages/cobbler/cobbler_collections/collection.py", line 262, in rename
self.listing.pop(oldname)
```
I can undo this change, but then I need to make changes to the rest of the code so that the error does not occur.
What is the best way to proceed in this case?

Reply:
Please keep the change but document in the name property of `item.py` that the name must be in lowercase. Please also add this hint to the Cobbler CLI documentation. This is very very important information imho.

Reply:
I think in fact we should try to prioritize this and make a split out PR (I would merge this of course before this PR so you can then rebase this PR on the new state of main that includes these changes).

Reply:
> Please keep the change but document in the name property of `item.py` that the name must be in lowercase. Please also add this hint to the Cobbler CLI documentation. This is very very important information imho.

A little different: a name can be in any case, but it is not allowed to have multiple names within the same collection that differ only in case. I'll add this hint to the documentation.

Reply:
> I think in fact we should try to prioritize this and make a split out PR (I would merge this of course before this PR so you can then rebase this PR on the new state of main that includes these changes).

Can we do this?
1. I am reverting changes in this PR that are not directly related to cache management
2. merge all commits into one
3. create a separate PR with bug fixes

Reply:
I will not enforce this plan as I believe that this will be a lot of effort but I would indeed prefer to see this. If you keep the commits clean I have nothing against it being in this PR.

Reply:
I have reverted this change for now, I will add it later as a separate commit.